src/scopes/activityReport/topic.js

```diff
@@ -21,10 +21,14 @@ const getTopicsSql = (topicsString) => `
     ON "ActivityReportObjectiveTopics"."topicId" = "Topics"."id"
     WHERE "Topics"."name" IN (${topicsString})`;
 
-export function withTopics(topics) {
-  const topicsString = topics.map((t) => `'${t}'`).join(',');
-  const arTopicsQuery = getArTopicsSql(topicsString);
-  const topicsQuery = getTopicsSql(topicsString);
+export function withTopics(topics, _options, _userId, validTopics) {
+  const validTopicsFromDb = validTopics ?? new Set();
```

Comment:
If `validTopics` is not provided and we instead define `validTopicsFromDb` as a `new Set()` with no members, then the following `topics.filter((t) => validTopicsFromDb.has(t))` will always return an empty array. Would it perhaps make sense then to immediately return the 'default' object if `validTopics` is null or undefined? Probably only save a microsecond of processing time, so not a big deal one way or the other.